// Copyright 2020 DeepMind Technologies Limited.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package courier;

import "courier/serialization/serialization.proto";

message CallRequest {
  // Name of the server method to be called.
  string method = 1;

  // Arguments for the method call.
  courier.CallArguments arguments = 2;
}

message CallResponse {
  // Result of the method call.
  courier.CallResult result = 2;
}

message ListMethodsRequest {}

message ListMethodsResponse {
  repeated string methods = 1;
}


service CourierService {
  // Calls a method on the server and returns the call's result.
  rpc Call(CallRequest) returns (CallResponse) {
  }

  // Lists the methods available on the server.
  rpc ListMethods(ListMethodsRequest) returns (ListMethodsResponse) {
  }

}
